agents-audit/src/main/java/org/apache/ranger/audit/utils/RangerJSONAuditWriter.java

```diff
@@ -108,16 +108,16 @@ public PrintWriter run()  throws Exception {
             // flush and check the stream for errors
             if (out.checkError()) {
                 // In theory, this count may NOT be accurate as part of the messages may have been successfully written.
-                // However, in practice, since client does buffering, either all of none would succeed.
+                // However, in practice, since client does buffer-ing, either all or none would succeed.
                 out.close();
-                closeWriter();
```

Comment:
Not calling `closeWriter()` here will result in subsequent call to `getLogFileStream()` to return the same `PrintWriter` closed in line #112 above - from `AbstractRangerAuditWriter.createWriter()`. Please review.

Reply:
Makes sense, updated the review, thank you for pointing it out!